File: app/src/main/java/com/vmenon/mpo/activity/BaseActivity.java
package com.vmenon.mpo.activity;

import androidx.appcompat.app.AppCompatActivity;

import com.vmenon.mpo.AppComponent;
import com.vmenon.mpo.MPOApplication;

public class BaseActivity extends AppCompatActivity {
    protected AppComponent getAppComponent() {
        return ((MPOApplication) getApplication()).getAppComponent();
    }
}

File: app/src/main/java/com/vmenon/mpo/activity/BaseActivity.kt
package com.vmenon.mpo.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import com.vmenon.mpo.AppComponent
import com.vmenon.mpo.MPOApplication

abstract class BaseActivity : AppCompatActivity() {
    protected lateinit var appComponent: AppComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        appComponent = (application as MPOApplication).appComponent
    }
}
